server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -496,7 +497,8 @@ public void onFailure(Throwable t)
                 }
               }
             }
-          }
+          },
+          Execs.directExecutor()
```

Comment:
`directExecutor()` and `sameThreadExecutor()` used by default are essentially the same, so how does it fix the problem?

Independent of that, there should be a code comment explaining why explicit executor is provided.